ipn/ipnlocal/local.go

```diff
@@ -3936,7 +3907,16 @@ func (b *LocalBackend) ShouldHandleViaIP(ip netip.Addr) bool {
 
 func (b *LocalBackend) LogoutSync(ctx context.Context) error {
 	b.mu.Lock()
+	if b.state == ipn.NeedsLogin {
```

Comment:
what about NoState? :)

Reply:
No can't do it on NoState, only on NeedsLogin. NeedsLogin is set when control has said we are logged out. NoState is when we don't know the state

Reply:
reworked to use presence of NodeKey instead